core/Updater.php

```diff
@@ -83,14 +83,52 @@ public function addUpdateObserver(UpdateObserver $listener)
      *
      * @param string $name The component name. Eg, a plugin name, `'core'` or dimension column name.
      * @param string $version The component version (should use semantic versioning).
+     * @param bool   $isNew indicates if the component is a new one (for plugins)
      */
-    public function markComponentSuccessfullyUpdated($name, $version)
+    public function markComponentSuccessfullyUpdated($name, $version, $isNew = false)
```

Comment:
Needed a new parameter to indicate if a new plugin was installed or updated, as that method is also used when installing new plugins